app/src/store/features/environmentVariables/slice.ts

```diff
@@ -0,0 +1,58 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { EnvironmentVariableValue } from "backend/environmentVariables/types";
+import { ReducerKeys } from "store/constants";
+import getReducerWithLocalStorageSync from "store/getReducerWithLocalStorageSync";
+
+const initialState = {
+  variables: {} as Record<string, Record<string, EnvironmentVariableValue>>,
+};
+
+const slice = createSlice({
+  name: ReducerKeys.ENVIRONMENT_VARIABLES,
+  initialState,
+  reducers: {
+    resetState: () => initialState,
+    setVariables: (
+      state,
+      action: PayloadAction<{
+        newVariables: Record<string, EnvironmentVariableValue>;
+        environment: string;
+      }>
+    ) => {
+      state.variables = {
+        ...state.variables,
+        [action.payload.environment]: {
+          ...state.variables[action.payload.environment],
+          ...Object.fromEntries(
+            Object.entries(action.payload.newVariables).map(([key, value]) => {
+              const prevValue = state.variables[action.payload.environment]?.[key];
+              return [
+                key,
+                {
+                  localValue: value.localValue ?? prevValue?.localValue,
+                  syncValue: value.syncValue ?? prevValue?.syncValue,
+                },
+              ];
+            })
+          ),
+        },
+      };
+    },
```

Comment:
Lets create the reducer file separately and user immer so that we don't need to spread the envValues